dtao.c

```diff
@@ -422,21 +424,40 @@ event_loop(void)
 	int ret;
 	int wlfd = wl_display_get_fd(display);
 
-	while (run_display) {
+	int tfd = 0;
+	struct itimerspec ts;
+	if (persist > 0) {
+		tfd = timerfd_create(CLOCK_MONOTONIC, TFD_CLOEXEC);
+		ts = (struct itimerspec) {.it_value.tv_sec = persist};
+	}
+	
+	while (run_display && !(eof_stdin && persist == 0)) {
 		fd_set rfds;
 		FD_ZERO(&rfds);
-		FD_SET(STDIN_FILENO, &rfds);
+		if (!eof_stdin)
+			FD_SET(STDIN_FILENO, &rfds);
+		else
+			if (persist > 0) {
+				timerfd_settime(tfd, 0, &ts, NULL);
+				FD_SET(tfd, &rfds);
+			}
+
 		FD_SET(wlfd, &rfds);
 
 		/* Does this need to be inside the loop? */
 		wl_display_flush(display);
 
-		ret = select(wlfd + 1, &rfds, NULL, NULL, NULL);
+		ret = select(MAX(tfd, wlfd) + 1, &rfds, NULL, NULL, NULL);
 		if (ret < 0)
 			EBARF("select");
 
-		if (FD_ISSET(STDIN_FILENO, &rfds))
-			read_stdin();
+		if (!eof_stdin) {
```

Comment:
No need to check `eof_stdin`, `FD_ISSET` will always be false if `eof_stdin` is false (is conditionally added)